### PR TITLE
Change datastore interface-wip

### PR DESCRIPTION
**Pull request by tarrantzhang**
This PR dedicates to reimplement datastore interface in elide. [Details](https://github.com/yahoo/elide/issues/294).

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java [hunk 1]
```diff
@@ -71,64 +77,56 @@ default void preCommit() {
...
+     * @param entity - the object to create in the data store.
+     * @param scope - contains request level metadata.
      */
-    <T> T loadObject(Class<T> entityClass, Serializable id);
+    void createObject(Object entity, RequestScope scope);
 
-    default <T> T loadObject(Class<T> entityClass, Serializable id, Optional<FilterExpression> filterExpression) {
-        return loadObject(entityClass, id);
+
+    default <T> T createNewObject(Class<T> entityClass) {
```
Comment (aklish) [inline elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java]: Remove and do this inside the persistent resource.

File: elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java [hunk 2]
```diff
@@ -199,18 +197,97 @@ default void preCommit() {
...
             String relationName,
-            Class<T> relationClass,
-            EntityDictionary dictionary,
             Optional<FilterExpression> filterExpression,
-            Sorting sorting,
-            Pagination pagination
-    ) {
-        return PersistentResource.getValue(entity, relationName, dictionary);
-    }
+            Optional<Sorting> sorting,
```
Comment (aklish) [inline elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java]: The default method for this should call relationTx.load with a FilterExpression on the parent entity ID.

File: elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java [hunk 3]
```diff
@@ -199,18 +197,97 @@ default void preCommit() {
...
+     * @param scope - contains request level metadata.
+     */
+    default void setAttribute(
+            Object entity,
+            String attributeName,
+            Object attributeValue,
+            RequestScope scope) {
+        Class<?> entityClass = entity.getClass();
+        EntityDictionary dictionary = scope.getDictionary();
+        String type = dictionary.getJsonAliasFor(entity.getClass());
```
Comment (aklish) [inline elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java]: I would leave default method here empty.

**Pull request closed without merging**